### PR TITLE
Add UISettings passthru

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description

Adds the required passthru cluster role + binding for UISettings  (actual RBAC is handled by the API Server).  Sim. to tiered policy.

Also, I do not believe we need the unauthenticated parts of the roles anymore.  I have removed these as a separate commit so that if it turns out we do (I tested and it seemed we didn't ... but just in case there is some odd corner case I haven't considered) then we can back out a single commit.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/apiserver.go [hunk 1]
```diff
@@ -1591,6 +1595,59 @@ func (c *apiServerComponent) tieredPolicyPassthruClusterRolebinding() *rbacv1.Cl
...
+// Calico Enterprise only
+func (c *apiServerComponent) uiSettingsPassthruClusterRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "tigera-uisettings-passthrough",
+		},
+		// If tiered policy is enabled we allow all authenticated users to access the main tier resource, instead
+		// restricting access using the tier.xxx resource type. Kubernetes NetworkPolicy and the
+		// StagedKubernetesNetworkPolicy are handled using normal (non-tiered) RBAC.
```
Comment (doublek) [inline pkg/render/apiserver.go:1610]: Stale comment.

**Review by doublek — APPROVED** [review]
One stale comment. Rest LGTM. No need to rereview.

**Review by doublek — APPROVED** [review]
👍

**Review by Brian-McM — APPROVED** [review]
